### PR TITLE
Chore: Replace acts_as_votable with our own likes feature.

### PR DESCRIPTION
**Pull request by KevinMulhern**
Because:
* It removes a dependency :tada:
* We were only utilizing a small part of the functionality acts_as_votable provides.
* Rolling our own allows for more flexibility and we can better model our domain - we can use "like" everywhere instead of having it in some places and using "votes" when acts_as_votable got involved.

This commit:
* Migrates the votes table to likes, drops unused columns and renames others to a more appropriate name.
* Adds a unique constraint to the likable and user - users should only ever be able to like a project submission once.
* Adds a likeable concern and shared examples so we can easily extend other records to be likeable in the future.
* Refactor the likes controller to act more like a toggle - it only needs an update action instead of create and destroy. This simplifies the component slightly too.
* Removes the acts_as_votable gem and model.

### CODE REVIEW

File: db/migrate/20230805091337_rename_votes_table_to_likes.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    rename_column :likes, :voter_id, :user_id
+
+    change_column_null :likes, :likeable_id, false
+    change_column_null :likes, :likeable_type, false
+    change_column_null :likes, :user_id, false
+
+    remove_column :likes, :voter_type
+    remove_column :likes, :vote_flag
+    remove_column :likes, :vote_scope
+    remove_column :likes, :vote_weight
```
Comment (rlmoser99) [inline db/migrate/20230805091337_rename_votes_table_to_likes.rb:16]: Wow, great timing as I wondered about the usage/need of columns this week.

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@
   it { is_expected.to have_many(:user_providers).dependent(:destroy) }
   it { is_expected.to have_many(:flags).dependent(:destroy) }
   it { is_expected.to have_many(:notifications) }
+  it { is_expected.to have_many(:notifications).dependent(:destroy) }
```
Comment (rlmoser99) [inline spec/models/user_spec.rb:18]: Is this supposed to be testing for likes?
```suggestion
  it { is_expected.to have_many(:likes).dependent(:destroy) }
```

**Review by rlmoser99 — APPROVED** [review]
Looks good to me. Just have one question on a test.

**Comment by ManonLef** [comment]
Nice! 
I am not sure if this expected behavior but when I upload a project, and then log in as Kevin to like and unlike, the heart doesn't change as shown in a screencapture. I also ran into a code 500 which may or may not be unrelated, which was after I removed my test project, and started liking and unliking test-user projects, screenshot attached right before the site crashed and showed me 500.
Screenshot 2023-08-05 at 22 02 16

https://github.com/TheOdinProject/theodinproject/assets/81025586/505d839d-1058-4e01-9e59-36cb0b47ebcb

**Comment by ManonLef** [comment]
This is now on your account with another project I uploaded through my test account. 
Likes number doesn't increase, heart only turns to outline after refresh. Old project submissions like behaviour seems to work as expected.
https://github.com/TheOdinProject/theodinproject/assets/81025586/224f3bdd-3044-44c8-9edc-579fa435f4a1

**Comment by rlmoser99** [comment]
Great catch @ManonLef. I tested it just using the test user's submissions, but didn't think about creating a new submission and then interacting with it using a different account.

**Comment by KevinMulhern** [comment]
Thanks both, top class reviewing 👏 

Looks like there was a bit of weirdness with using [`first_or_create`](https://github.com/TheOdinProject/theodinproject/pull/4068/commits/b6a703a969b229269d1a73d96a67c8b629f58a69#diff-a0bed38ab2316aaf6bf44c9d28ab109b72a812739f2a47a4bc8484937e0e332dL14). It didn't appear to be reloading the project submission object with an updated like count when that was used. Should hopefully all be working now 🙏

**Review by rlmoser99 — APPROVED** [review]
Looks good to me. Like to new test coverage! 💪

**Pull request merged**